cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -150,21 +150,27 @@ int64_t kafka_consumer::get_committed_offset(std::string const &topic, int parti
                "Failed retrieve Kafka committed offsets");
 
   int64_t offset = toppar_list[0]->offset();
-  return offset > 0 ? offset : 0;
+  return offset > 0 ? offset : -1001;
 }
 
-std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics() const
+std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics(
+  std::string specific_topic) const
 {
-  auto const md = [&]() {
+  auto const metadata = [&]() {
+    std::string errstr;
+    auto spec_topic = std::unique_ptr<RdKafka::Topic>(
+      RdKafka::Topic::create(consumer.get(), specific_topic, nullptr, errstr));
+
     RdKafka::Metadata *md;
-    CUDF_EXPECTS(RdKafka::ERR_NO_ERROR == consumer->metadata(true, nullptr, &md, default_timeout),
+    CUDF_EXPECTS(RdKafka::ERR_NO_ERROR ==
+                   consumer->metadata(
+                     spec_topic == nullptr ? true : false, spec_topic.get(), &md, default_timeout),
```

Comment:
Just realized this is completely redundant. Could be:
```suggestion
                     spec_topic == nullptr, spec_topic.get(), &md, default_timeout),
```